### PR TITLE
Fix CLI commands

### PR DESCRIPTION
**Pull request by anjastrunk**
There is a mistake in documentation. Calling gx-credential-generator from CLI as follows 

```shell
python3 -m generator.cli csp
```

results in the following error:

```shell
 python3 -m generator.cli csp --config config.yaml --auto-sign
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/anja-strunk/Repos/SCS/gx-credential-generator/cli.py", line 223, in <module>
    def create_vmso_vcs(conf: Config, cloud: str, csp_vcs: List[dict], timeout: int = 12) -> dict[dict]:
TypeError: 'type' object is not subscriptable
```

### CODE REVIEW

**Comment by anjastrunk** [comment]
Additionally to bug fix, I added the following minor changes:
- add flavor name
- capitalize config key "gxdch"

File: README.md [hunk 1]
```diff
@@ -91,7 +91,7 @@ gx-credential-generator requires some configuration options. See [configuration]
 Create Gaia-X Credential for a CSP without specifying a configuration file. This implies the default path at `/etc/gx-credential-generator/config.yaml`, which must exist:
 
 ```commandline
-python3 -m generator.cli csp
+python3 -m cli csp
```
Comment (mbuechse) [inline README.md:94]: I haven't understood why this change was necessary. The description of this PR didn't help me, because it showed two different invokations, where the first one didn't include the error, and with the second one, cli.py had already been moved out of the `generator` directory.

I could not reproduce the error with the main branch, but I faced another problem: I don't have a private key yet, and the docs don't tell me what kind of key is required (actually, it would be nice to provide a command-line).

File: cli.py [hunk 2]
```diff
@@ -36,11 +36,11 @@
...
 
 VC_NAME_LOOKUP = {
-    "lp": "Legal Person",
+    "legal_person": "Legal Person",
     "lrn": "Legal Registration Number",
     "tandc": "Gaia-X Terms and Conditions",
-    "cs_csp": "GXDCH Compliance Service",
-    "cs_so": "GXDCH Compliance Service",
+    "csp_compliance": "GXDCH Compliance Service",
+    "so_compliance": "GXDCH Compliance Service",
```
Comment (mbuechse) [inline cli.py:43]: Is it intentional that both keys are mapped to the same value? Maybe this warrants a comment.

File: generator/common/const.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 """
 Constants used by SCS GX Credential Generator.
 """
-CONFIG_FILE = "config/config.yaml"
+CONFIG_FILE = "config/config.yaml.template"
```
Comment (mbuechse) [inline generator/common/const.py:4]: That doesn't seem quite right. The template is not usable.

(Therefore it is correct to rename the file, but the code should still refer to a proper config file, not just a template.)

File: cli.py [hunk 4]
```diff
@@ -247,7 +248,7 @@ def create_vmso_vcs(conf: Config, cloud: str, csp_vcs: List[dict], timeout: int
     vmso_vc = {
         '@context': [const.VC_CONTEXT, const.JWS_CONTEXT, const.REG_CONTEXT],
         'type': "VerifiableCredential",
-        'id': cred_settings[const.CONFIG_CRED_BASE_CRED_URL] + "/vmo.json",
+        'id': cred_settings[const.CONFIG_CRED_BASE_CRED_URL] + "/vmso.json",
```
Comment (mbuechse) [inline cli.py:251]: Why this change? And why was `vmo.json` added to the source tree?

**Review by mbuechse — COMMENTED** [review]
I'm a bit baffled by the changes. I'm afraid I need some more context. Please see my comments.

File: README.md (hunk 1 above)
Comment (anjastrunk) [reply README.md:94]: > ... because it showed two different invokations

You are right, there was a mistake in the description, which I corrected

Comment (anjastrunk) [reply README.md:94]: > I could not reproduce the error with the main branch, but I faced another problem: I don't have a private key yet, and the docs don't tell me what kind of key is required (actually, it would be nice to provide a command-line).

This tool generates Gaia-X Credentials. Requirements to private key are described in [Gaia-X documents](https://docs.gaia-x.eu/) and in our upcoming blog post "[Demystifying Gaia-X Credentials](https://staging.scs.community/website/de/2024/09/20/demystifying-gaia-x-credentials/)". If you want to run credential generator, you have to set up whole infrastructure for Gaia-X Credentials as described in blog post.

File: cli.py (hunk 2 above)
Comment (anjastrunk) [reply cli.py:43]: Yes. Keys are mapping to strings used in log statements.

File: generator/common/const.py (hunk 3 above)
Comment (anjastrunk) [reply generator/common/const.py:4]: You are right. This change was not intended and I deteted it.

File: cli.py (hunk 4 above)
Comment (anjastrunk) [reply cli.py:251]: > Why this change? 

The idea is to publish generated credentials at URL given in credential's `id` element. This change was necessary, to align file name with the id of Gaia-X Credential for `VirtualMachineServiceOffering`.

> And why was vmo.json added to the source tree?

This was not intended and I removed `vmo.json`.

File: README.md (hunk 1 above)
Comment (mbuechse) [reply README.md:94]: I'm sorry, but no, the problem had nothing to do with the folder, as far as I can reconstruct from your command-line output. The actual problem was the following:

1. You used the `dict[dict]` type annotation (which this PR also removes).
2. You used Python 3.8, which does not support that annotation.

Even with Ubuntu 22.04, you should have Python 3.10.

Comment (anjastrunk) [reply README.md]: You are right. I worked with python 3.10. I wasn't aware, that this was the problem. Thank's for pointing this out. I reverted to move cli.py out and added a hint to use python 3.10 to documentation.

**Review by mbuechse — APPROVED** [review]
LGTM
